common/djangoapps/enrollment/data.py

```diff
@@ -88,7 +107,18 @@ def get_course_enrollment(username, course_id):
         enrollment = CourseEnrollment.objects.get(
             user__username=username, course_id=course_key
         )
-        return CourseEnrollmentSerializer(enrollment).data
+
+        # Edraak: use EdraakCourseEnrollmentSerializer to serialize enrollments
+        if request is None:
+            data = CourseEnrollmentSerializer(enrollment).data
+        else:
+            data = EdraakCourseEnrollmentSerializer(
+                enrollment,
+                context={'request': request}
```

Comment:
Please use the `request_cache.get_current_request()` instead of passing the request all around the place.

```suggestion
                context={'request': request_cache.get_request_or_stub()}
```

Also you need to `import request_cache`

Reply:
Done